src/App.js

```diff
@@ -26,67 +29,54 @@ import NotFound from './components/NotFound';
-      // trees: [],
-      // stars: [],
-      query: ''
+      query: 'mountains'
+    
     };
   } 
 
-
-componentDidMount() {
-  this.querySearch();
-  console.log("component mounted")
-}
-
-
-querySearch = (query = 'mountains') => {
-
-    axios.get(`https://www.flickr.com/services/rest/?method=flickr.photos.search&api_key=${apiKey}&tags=${query}&per_page=24&format=json&nojsoncallback=1`)
-      .then(response => { //DOESN'T STOP UPDATING WITH NEW PHOTOS
-         this.setState(
-          {photos: response.data.photos.photo,
-            query: `${query}`},
-         // console.log("state w axios:", this.state)
-          )
-    })  
-      .catch(error => {
-        console.log('Error fetching and parsing data:', error);
-      });
-  }
+  componentDidMount() {
+    this.querySearch();
+    console.log("component mounted");
+  } 
+ 
+  querySearch = (query =this.state.query) => {
+      axios.get(`https://www.flickr.com/services/rest/?method=flickr.photos.search&api_key=${apiKey}&tags=${query}&per_page=24&format=json&nojsoncallback=1`)
+        .then(response => { 
+          this.setState({
+            photos: response.data.photos.photo,
+            query: `${query}`
+          })
+      })  
+        .catch(error => {
+          console.log('Error fetching and parsing data:', error);
+        });
+    }
 
   render() { 
     return (
       <BrowserRouter>
-    <div>
-        <div className="main-header">
-          <div className="inner">
-            <h1 className="main-title">Photo Search</h1>
+        <div>
+          <div className="main-header">
+            <div className="inner">
+              <h1 className="main-title">Photo Search</h1>
+          </div>
         </div>
-      </div>
-      <Search onSearch={this.querySearch}/> 
+        <Search onSearch={this.querySearch}/> 
       </div>  
-      <div className='main-nav'>
-          <Nav data={this.querySearch}/> 
-      </div>
-      <div className="main-content">
-          <Switch> 
-              <Route exact path ="/" render={() => <Redirect to="/search/mountains" />} /> 
-              <Route path="/search/mountains" render={() => <PhotoContainer data={this.state} />}/>
-              <Route path="/search/trees" render={() => <PhotoContainer data={this.state} />}/>
-              <Route path="/search/stars" render={() => <PhotoContainer data={this.state} />}/>
-              <Route path={"/search/:query"} render= {() => <PhotoContainer data={this.state}/>}/>
-              <Route path="/no-results" render={() => <NotFound data={this.state}/>} />
-          </Switch>  
-      
-      </div>      
-        
-    
+        <div className='main-nav'>
+            <Nav data={this.querySearch}/> 
+        </div>
+        <div className="main-content">
+            <Switch> 
+                <Route exact path ="/" render={() => <Redirect to="/search/?q=mountains" />} /> 
+                <Route path="/search" render={() => <PhotoContainer data={this.state} />}/>
```

Comment:
the prop here `data` should be more descriptive.

For instance, photos={this.state.photos}